### PR TITLE
[bug](s3) fix S3 file system gets absolute path

### LINKED ISSUE #44902: [Bug] Can't list S3 objects if the root path of S3 resource has a leading slash
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

2.0

### What's Wrong?

If we create a S3 resource and whe `s3.root.path` has a leading slash(/), such as: `/root`, the `S3Client:ListObjectsV2` will not work in `s3_file_system.cpp`.
See more in : https://stackoverflow.com/questions/52443839/s3-what-exactly-is-a-prefix-and-what-ratelimits-apply and https://docs.aws.amazon.com/AmazonS3/latest/userguide/using-prefixes.html

### What You Expected?

`S3Client:ListObjectsV2` gets work even though `s3.root.path` has a leading slash

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by DarvenDuan**
### What problem does this PR solve?

Issue Number: close https://github.com/apache/doris/issues/44902

Related PR: #xxx

###  Problem Summary
If we create a S3 resource for cooldown data storage and the s3.root.path has a leading slash(/), such as: /root, the remote S3 data file will not be deleted if we drop the cooldown tablet from Doris.

### Reason
Doris will get the path of all files of a tablet and then delete those files, The AWS `S3Client:ListObjectsV2` uses objects prefix to get the object files, but if the prefix has a leading slash(/), the`S3Client:ListObjectsV2` gets empty result.

### Solution
use _prefix instead of _root_path in S3FileSystem for getting absolute path, _prefix is normalized in constructor and it is removed the first and last '/'.

### test case
1. create a s3 resource with a leading '/' and a storage policy base on it:
```
CREATE RESOURCE "test_resource"
PROPERTIES
(
    "type" = "s3",
    "s3.endpoint" = "xxx",
    "s3.region" = "xxx",
    "s3.bucket" = "xxx",
    "s3.root.path" = "/tmp",
    "s3.access_key" = "xx",
    "s3.secret_key" = "xx"
);

CREATE STORAGE POLICY test_policy PROPERTIES (
    "storage_resource" = "test_resource",
    "cooldown_ttl" = "1"
)
```
2. create a table and set the storage_policy to `test_policy` and insert test data:
```
CREATE TABLE `test_table` (
  `k` bigint,
  `v` bigint
) ENGINE=OLAP
DUPLICATE KEY(`k`)
DISTRIBUTED BY HASH(`k`) BUCKETS 1
PROPERTIES (
"replication_num" = "1",
"storage_policy" = "test_policy"
);

insert into test_table values (1,2),(2,3);
```
3. wait for the rowset cooldown to s3.
4. truncate test table
```
truncate table test_table force;
```
5. If we get a log like ` delete remote rowsets of tablet. root_path="/tmp", tablet_id=xxx` in be.INFO, the remote tablet file should be deleted.

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [x] Unit Test
    - [x] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: be/src/io/fs/s3_file_system.h [hunk 1]
```diff
@@ -121,7 +121,7 @@ class S3FileSystem final : public RemoteFileSystem {
             abs_path = path;
         } else {
             // path with no schema
-            abs_path = _root_path / path;
+            abs_path = _prefix / path;
```
Comment (DarvenDuan) [inline be/src/io/fs/s3_file_system.h:124]: _prefix is normalized in S3FileSystem's constructor, removed the first and last '/'

**Comment by DarvenDuan** [comment]
run buildall

**Comment by DarvenDuan** [comment]
run cloud_p0

**Review by caiconghui — APPROVED** [review]

**Review by GoGoWen — APPROVED** [review]
LGMT

**Pull request merged**